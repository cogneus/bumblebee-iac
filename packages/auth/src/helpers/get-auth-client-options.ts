import { ClientOptions } from 'bumblebee-common'
import { env } from 'process'

export const getAuthClientOptions = (): ClientOptions => {
  const { RESOURCE_NAME_PREFIX, SECRET_ARN_PREFIX, KMS_ID, SECRET_SIGNING_KEY_NAME } = env
  return {
    secrets: {
      idPrefix: SECRET_ARN_PREFIX,
      prefix: RESOURCE_NAME_PREFIX,
      names: {},
      ids: {
        authSigningKey: { name: SECRET_SIGNING_KEY_NAME },
      },
      kmsId: KMS_ID,
    },
  }
}
