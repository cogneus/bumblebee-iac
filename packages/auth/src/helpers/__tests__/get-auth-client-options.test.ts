import { getAuthClientOptions } from '../get-auth-client-options'

describe('Get auth client options tests', () => {
  beforeEach(() => {
    jest.clearAllMocks()
    process.env = {
      AWS_REGION: 'eu-west-1',
      LOG_LEVEL: 'debug',
      NODE_ENV: 'dev',
      RESOURCE_NAME_PREFIX: 'test-prefix',
      SECRET_SIGNING_KEY_NAME: 'signingKey',
      TOKEN_AUDIENCE: 'audience',
      TOKEN_EXPIRY: 'expiry',
      TOKEN_ISSUER: 'issuer',
    } as any
  })

  test('should return the value of the secrets key', async () => {
    const value = getAuthClientOptions()
    expect(value).toMatchSnapshot()
  })
})
