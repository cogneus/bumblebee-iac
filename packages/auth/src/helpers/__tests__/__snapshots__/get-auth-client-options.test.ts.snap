// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get auth client options tests should return the value of the secrets key 1`] = `
{
  "secrets": {
    "idPrefix": undefined,
    "ids": {
      "authSigningKey": {
        "name": "signingKey",
      },
    },
    "kmsId": undefined,
    "names": {},
    "prefix": "test-prefix",
  },
}
`;
